resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicExecutorFeatureStep.scala

```diff
@@ -172,7 +184,7 @@ private[spark] class BasicExecutorFeatureStep(
         .addToImagePullSecrets(kubernetesConf.imagePullSecrets(): _*)
         .endSpec()
       .build()
-    SparkPod(executorPod, containerWithLimitCores)
+    SparkPod(executorPod, containerWithLimitGpus)
```

Comment:
What if you want to request both GPU's and another resource (e.g., FPGA's) for your executors? You'd have to hard-code chain another optional map like you did in the pull requests. Your current code does not pave a good path forward to requesting other types of resources.

Reply:
By that logic, it might make sense to make separate steps for varying types of resources. Might be overloading the logic handled by `BasicExecutorStep`. @mccheah for comment

Reply:
Yes ... not in love with the way this is currently structured.